WalletWasabi.Gui/CommandLine/Daemon.cs

```diff
@@ -95,31 +109,31 @@ internal async Task RunAsync(string walletName, bool mixAll, bool keepMixAlive)
 					}
 
 					await Task.Delay(3000);
+
 					if (Global.KillRequested)
 					{
 						break;
 					}
 
-					bool anyCoinsQueued = WalletService.ChaumianClient.State.AnyCoinsQueued();
-					if (!anyCoinsQueued && keepMixAlive) // If no coins queued and mixing is asked to be kept alive then try to queue coins.
+					// If no coins enqueued then enqueue the large anonset coins and mix to another wallet.
+					if (isDestinationSpecified && !AnyCoinsQueued())
 					{
-						// Don't do mixall here, the mixall says all the coins has to be mixed once, it doesn't says it has to be requeued all the time.
-						await TryQueueCoinsToMixAsync(mixAll: false, password);
+						WalletService.ChaumianClient.DestinationKeyManager = destinationKeyManager;
+						await TryQueueCoinsToMixAsync(password, minAnonset: WalletService.ServiceConfiguration.MixUntilAnonymitySet);
 					}
 
-					if (Global.KillRequested)
+					// If no coins were queued then try to queue coins those have less anonset and mix it into the same wallet.
+					if (!AnyCoinsQueued())
 					{
-						break;
+						// Don't do mixall here, the mixall says all the coins has to be mixed once, it doesn't says it has to be requeued all the time.
+						WalletService.ChaumianClient.DestinationKeyManager = WalletService.ChaumianClient.KeyManager;
+						await TryQueueCoinsToMixAsync(password, maxAnonset: WalletService.ServiceConfiguration.MixUntilAnonymitySet - 1);
 					}
```

Comment:
I would write it this way to not check `if (!AnyCoinsQueued())` twice:
```
if (!AnyCoinsQueued())
{
	if (isDestinationSpecified)
	{
		WalletService.ChaumianClient.DestinationKeyManager = destinationKeyManager;
		await TryQueueCoinsToMixAsync(password, minAnonset: WalletService.ServiceConfiguration.MixUntilAnonymitySet);
	}

	// If no coins were queued then try to queue coins those have less anonset and mix it into the same wallet.
	// Don't do mixall here, the mixall says all the coins has to be mixed once, it doesn't says it has to be requeued all the time.
	WalletService.ChaumianClient.DestinationKeyManager = WalletService.ChaumianClient.KeyManager;
	await TryQueueCoinsToMixAsync(password, maxAnonset: WalletService.ServiceConfiguration.MixUntilAnonymitySet - 1);
}
```

Reply:
You have to check that as `TryQueueCoinsToMixAsync` may change the result.

Reply:
Was about to say that my suggestion is correct only if `TryQueueCoinsToMixAsync` doesn't change the state. Thanks